lab-segmentation.cpp

```diff
@@ -88,130 +100,148 @@ void lab11()
     }
 
     // Extract pixel features from the image.
-    cv::Mat feature_image = extractFeatures(frame);
-
-    if (model)
-    {
-      // Compute how well the pixel features fit with the model.
-      cv::Mat mahalanobis_img = model->computeMahalanobisDistances(feature_image);
-      cv::imshow(win_name_result, mahalanobis_img);
-
-      // Segment out the areas of the image that fits well enough.
-      cv::Mat segmented = performSegmentation(mahalanobis_img, current_threshold, use_otsu);
-      cv::setTrackbarPos(threshold_trackbar_name, win_name_input, current_threshold);
-
-      // Set segmented area to green.
-      frame.setTo(cv::Scalar{0, 255, 0}, segmented);
-    }
+    feature_image = extractFeatures(frame);
 
     // Update if using adaptive model.
-    if (model && use_adaptive_model)
+    if (use_adaptive_model)
     {
       cv::Mat new_samples = extractTrainingSamples(feature_image, sampling_rectangle);
       updateSamples(current_samples, new_samples, adaptive_update_ratio);
-      model = std::make_shared<MultivariateNormalModel>(current_samples);
+      model = ModelType(current_samples);
     }
 
+    // Compute how well the pixel features fit with the model.
+    cv::Mat mahalanobis_img = model.computeMahalanobisDistances(feature_image);
+
+    // Segment out the areas of the image that fits well enough.
+    cv::Mat segmented = performSegmentation(mahalanobis_img, current_threshold, use_otsu, max_std_dev);
+
+    // Set segmented area to green.
+    frame.setTo(cv::Scalar{0, 255, 0}, segmented > 0);
+
     // Draw current frame.
     drawSamplingRectangle(frame, sampling_rectangle);
+
+    cv::Mat viz;
+    cv::normalize(mahalanobis_img, viz, 1.0, 0.0, cv::NORM_MINMAX);
```

Comment:
Trenger vi dette?

Reply:
Alt over 1 blir i hvert fall hvitt/"utbrent", men kan dele på max sånn som du har gjort i python.

Reply:
Normaliserer [0, 1]

![image](https://user-images.githubusercontent.com/11663739/153410825-405d6e62-73e8-49d2-bf65-ff9edc018d8d.png)

Deler på max

![image](https://user-images.githubusercontent.com/11663739/153411026-38f392ae-246e-4a96-9717-e5b76223650f.png)

Hva tenker du? Det er forsåvidt nice to know at ting er utenfor max threshold, men på den annen side er det penere å se på det normaliserte bildet.

Reply:
Da er bildene hele tiden sammenlignbare, uansett hva som er i bildet